src/platforms/android/server/display.cpp

```diff
@@ -359,9 +359,9 @@ bool mga::Display::apply_if_configuration_preserves_display_buffers(
     if (!config.external().connected || displays.display_present(mga::DisplayName::external))
     {
         configure_locked(conf, lock);
-        return true;
+        return false;
     }
-    return false;
+    return true;
```

Comment:
Won't this result in the compositor incorrectly thinking that the display configuration has been applied?